source/databricks/calculation-engine/tests/calculation/wholesale/test_tariff_calculators.py

```diff
@@ -339,34 +340,179 @@ def test__calculate_tariff_price_per_ga_co_es__rounds_total_amount_correctly(
+) -> None:
+    # Arrange
+    rows = [
+        _create_tariff_hour_row(charge_time=datetime(2020, 1, 1, 0)),
+        _create_tariff_hour_row(charge_time=datetime(2019, 12, 31, 23)),
         _create_tariff_hour_row(charge_time=datetime(2019, 12, 31, 22)),
     ]
     tariffs = spark.createDataFrame(data=rows, schema=tariff_schema)
 
     # Act
     actual = sum_within_month(
-        calculate_tariff_price_per_ga_co_es(tariffs), "Europe/Copenhagen"
+        calculate_tariff_price_per_ga_co_es(tariffs), DEFAULT_TIME_ZONE
     )
 
     # Assert
     assert actual.collect()[0][Colname.total_amount] == Decimal("2.010005")
-    assert actual.collect()[1][Colname.total_amount] == Decimal("8.040020")
-    assert actual.collect()[2][Colname.total_amount] == Decimal("2.010005")
-    assert actual.collect()[1][Colname.qualities] == ["calculated", "estimated"]
+    assert actual.collect()[1][Colname.total_amount] == Decimal("4.020010")
     assert actual.collect()[1][Colname.charge_time] == datetime(2019, 12, 31, 23)
-    assert actual.count() == 3
+    assert actual.count() == 2
+
+
+def test__sum_within_month__daylight_savings_time_summertime(
+    spark: SparkSession,
+) -> None:
+    # Arrange
+    rows = [
+        _create_tariff_hour_row(charge_time=datetime(2020, 3, 31, 23)),
+        _create_tariff_hour_row(charge_time=datetime(2020, 3, 31, 22)),
+        _create_tariff_hour_row(charge_time=datetime(2020, 3, 31, 21)),
+    ]
+    tariffs = spark.createDataFrame(data=rows, schema=tariff_schema)
+
+    # Act
+    actual = sum_within_month(
+        calculate_tariff_price_per_ga_co_es(tariffs), DEFAULT_TIME_ZONE
+    )
+
+    # Assert
+    assert actual.collect()[0][Colname.total_amount] == Decimal("2.010005")
+    assert actual.collect()[1][Colname.total_amount] == Decimal("4.020010")
+    assert actual.collect()[0][Colname.charge_time] == datetime(2020, 3, 31, 21)
+    assert actual.collect()[1][Colname.charge_time] == datetime(2020, 3, 31, 22)
+    assert actual.count() == 2
+
+
+def test__sum_within_month__daylight_savings_time_wintertime(
+    spark: SparkSession,
+) -> None:
+    # Arrange
+    rows = [
+        _create_tariff_hour_row(charge_time=datetime(2020, 10, 31, 23)),
+        _create_tariff_hour_row(charge_time=datetime(2020, 10, 31, 22)),
+        _create_tariff_hour_row(charge_time=datetime(2020, 10, 31, 21)),
+    ]
+    tariffs = spark.createDataFrame(data=rows, schema=tariff_schema)
+
+    # Act
+    actual = sum_within_month(
+        calculate_tariff_price_per_ga_co_es(tariffs), DEFAULT_TIME_ZONE
+    )
+
+    # Assert
+    assert actual.collect()[0][Colname.total_amount] == Decimal("4.020010")
+    assert actual.collect()[1][Colname.total_amount] == Decimal("2.010005")
+    assert actual.collect()[0][Colname.charge_time] == datetime(2020, 10, 31, 21)
+    assert actual.collect()[1][Colname.charge_time] == datetime(2020, 10, 31, 23)
+    assert actual.count() == 2
+
+
+def test__sum_within_month__creates_sum_total_quantity(
```

Comment:
Perhaps align naming of these "create sum" test cases and put them next to each other?
```suggestion
def test__sum_within_month__sums_quantity_per_month(
```